packages/trpc/server/routers/viewer/slots.tsx

```diff
@@ -151,12 +155,48 @@ export async function getSchedule(
       },
       users: {
         select: {
-          username: true,
+          // username: true,
           ...availabilityUserSelect,
         },
       },
     },
   });
+
+  const isDynamicBooking = !input.eventTypeId;
+  // For dynamic booking, we need to get and update user credentials, schedule and availability in the eventTypeObject as they're required in the new availability logic
+  const dynamicEventType = getDefaultEvent(input.eventTypeSlug);
+  let dynamicEventTypeObject = dynamicEventType;
+
+  if (isDynamicBooking) {
+    const usernameList = input.usernameList;
+    const users = await ctx.prisma.user.findMany({
+      where: {
+        username: {
+          in: usernameList as string[],
```

Comment:
Can this be inferred? I see that it's coming from `input` which should be already type from a zod schema

Reply:
Since the usernameList can be a null array as well, if we don't do this, it'll throw a type error stating `Type 'null' is not assignable to type 'string'`, even though it'll only ever be called when usernameList isn't null. This just clears out that type error.

Reply:
> I see that it's coming from input which should be already typed from a zod schema

So yeah, and the zod schema has it as nullable array, hence the typecasting

Reply:
But... do we accept null arrays as inputs? If we don't let's make it not optional or non-nullable. That's the point of validating. We need to check or filter our nulls anyways don't we?